driver-core/src/test/functional/com/mongodb/client/WithWrapper.java

```diff
@@ -66,15 +64,6 @@ default WithWrapper withSystemProperty(final String name, final String value) {
     }
 
     static Map<String, String> getEnvInnerMap() {
-        try {
-            Map<String, String> env = getenv();
-            Field field = env.getClass().getDeclaredField("m");
-            field.setAccessible(true);
-            @SuppressWarnings("unchecked")
-            Map<String, String> result = (Map<String, String>) field.get(env);
-            return result;
-        } catch (IllegalAccessException | NoSuchFieldException e) {
-            throw new RuntimeException(e);
-        }
+        return FaasEnvironmentAccessor.getFaasEnvMap();
```

Comment:
This looks good.  Only change i would make is to inline this method, since the name no longer makes sense.